lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -555,6 +558,9 @@ data ApiTransaction (n :: NetworkDiscriminant) = ApiTransaction
     , outputs :: ![AddressAmount (ApiT Address, Proxy n)]
     , withdrawals :: ![ApiWithdrawal n]
     , status :: !(ApiT TxStatus)
+    -- fixme: #2073 not sure whether empty metadata should be null or missing
+    -- key.
+    , metadata :: !(Maybe (ApiT TxMetadata))
```

Comment:
My own preference is to have a `null` value so that it is more explicit client side but I admit not being 100% sure about how the default aeson generic options treat this in our API at the moment.

Reply:
That is also my preference actually. But when submitting a transaction, it is more convenient for users if they can omit the key. Our default generic options are to omit Nothing fields.

Reply:
Will sort it out in #2089.